cmake/CMakeLists.txt

```diff
@@ -230,14 +230,18 @@ if (onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS)
   add_compile_definitions(ENABLE_LANGUAGE_INTEROP_OPS)
 endif()
 
-if (NOT (onnxruntime_ENABLE_PYTHON AND onnxruntime_ENABLE_TRAINING))
+if (NOT (onnxruntime_ENABLE_PYTHON AND onnxruntime_ENABLE_TRAINING AND (NOT onnxruntime_BUILD_SHARED_LIB)))
```

Comment:
Let's make ENABLE_LANGUAGE_INTEROP_OPS to Linux only for now. I worry it doesn't work well in Windows Debug build because you undef _DEBUG in some of the source files. 
If there is a need to make it work on Windows, let's carefully examine how to solve the inconsistency, and if Ryan's debug allocator still works.

Reply:
Yeah, I spent days solving a problem around Win Debug/Relase build. That should work. But as you suggested, I will limit this to Linux because we don't have requirement for other platform.

Reply:
resolved